microsoft/testsuites/mdatp/mdatp.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import pathlib
+
+from assertpy import fail
+
+from lisa import Node, SkippedException, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Windows
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import simple_requirement
+
+# some constants from  check-mdatp.sh
+# if any mdatp install in /etc/opt is found
+EXIT_MDATP_AGENT_INSTALLED = 1
+# if mdatp az extension is installed
+EXIT_MDE_INSTALLED = 2
+# if any log dirs are found
+EXIT_MDATP_LOGS_FOUND = 4
+# if an installation log is found
+EXIT_MDATP_INSTALL_LOGS_FOUND = 8
+# if an onboarding blob is found
+EXIT_ONBOARD_INFO_FOUND = 16
+
+
+@TestSuiteMetadata(
+    area="mdatp",
+    category="functional",
+    description="""
+        Test to verify there are no pre-installed copies of mdatp.
+    """,
+)
+class MdatpSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+           Check for mdatp endpoint/cloud install, dump config info.
+           Fails if mdatp is installed in the image.
+           Raises specific error messages depending on the type of info
+           found
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            unsupported_os=[Windows], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_mdatp_not_preinstalled(self, node: Node) -> None:
+        # collect some paths before we start the test
+        checker = "check-mdatp.sh"
+        local_path = pathlib.PurePath(__file__).parent.joinpath(checker)
+        working_path = node.get_working_path()
+        script_path = working_path.joinpath(checker)
+
+        # copy the bash script to the node
+        node.shell.copy(local_path=local_path, node_path=script_path)
+        result = node.execute(
+            cmd=f"chmod +x {str(script_path)}",
+            shell=True,
+            sudo=True,
+        )
+        # and run the script to check and dump defender info
+        result = node.execute(
+            cmd=f"{str(script_path)}",
+            shell=True,
+            sudo=True,
+            no_debug_log=True,
+        )
+        script_output = result.stdout.strip()
+        exit_code = result.exit_code
+        if exit_code is None:
```

Comment:
it's impossible to be None from LISA logic. If it needs, assert is not None.

Reply:
fixed, thanks!